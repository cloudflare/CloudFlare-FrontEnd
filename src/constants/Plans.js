const LOCALIZED_PRO_PLAN_ID = 'constants.plans.pro';
const LOCALIZED_BIZ_PLAN_ID = 'constants.plans.biz';
const LOCALIZED_ENT_PLAN_ID = 'constants.plans.ent';

export const FREE_PLAN = 'free';
export const PRO_PLAN = 'pro';
export const BIZ_PLAN = 'business';
export const ENT_PLAN = 'enterprise';

export function planNeedsUpgrade(currentPlan, minimumPlan) {
  var planList = {};
  planList[FREE_PLAN] = 0;
  planList[PRO_PLAN] = 1;
  planList[BIZ_PLAN] = 2;
  planList[ENT_PLAN] = 3;

  return planList[currentPlan] < planList[minimumPlan];
}

export function getLocalizedPlanId(planName) {
  var localizedPlanName = planName;
  switch (planName) {
    case PRO_PLAN:
      localizedPlanName = LOCALIZED_PRO_PLAN_ID;
      break;
    case BIZ_PLAN:
      localizedPlanName = LOCALIZED_BIZ_PLAN_ID;
      break;
    case ENT_PLAN:
      localizedPlanName = LOCALIZED_ENT_PLAN_ID;
      break;
    default:
      // This should never happen
      break;
  }

  return localizedPlanName;
}
