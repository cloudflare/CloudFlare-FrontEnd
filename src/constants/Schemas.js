import { normalize, Schema, arrayOf } from 'normalizr';

const zoneSchema = new Schema('zones', { idAttribute: 'name' });
const zoneRailgunSchema = new Schema('railguns', { idAttribute: 'id' });

export function normalizeZoneByIdGetAll(zoneId, result) {
  var zoneSchemaById = new Schema(zoneId, { idAttribute: 'id' });
  return normalize(result, arrayOf(zoneSchemaById));
}

export function normalizeZoneGetAll(result) {
  return normalize(result, arrayOf(zoneSchema));
}

export function normalizeZoneRailgunGetAll(result) {
  return normalize(result, arrayOf(zoneRailgunSchema));
}
