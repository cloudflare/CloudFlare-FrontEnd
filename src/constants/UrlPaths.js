export const ANALYTICS_PAGE = '/analytics';
export const CLOUDFLARE_API_KB_ARTICLE_PAGE = 'https://support.cloudflare.com/hc/en-us/articles/200167836-Where-do-I-find-my-CloudFlare-API-key-';
export const CLOUDFLARE_FORGOT_PASSWORD_PAGE = 'https://cloudflare.com/a/forgot-password';
export const CLOUDFLARE_SIGNUP_PAGE = 'https://www.cloudflare.com/a/sign-up';
export const CLOUDFLARE_DASHBOARD_PAGE = 'http://cloudflare.com/a/overview/';
export const CLOUDFLARE_UPGRADE_PAGE = 'http://cloudflare.com/a/upgrade-plan';
export const DOMAINS_OVERVIEW_PAGE = '/zones';
export const HOME_PAGE = '/home';
export const MORE_SETTINGS_PAGE = '/more-settings';
export const LOGIN_PAGE = '/login';
export const SIGN_UP_PAGE = '/sign-up';
export const SUPPORT_PAGE = 'https://support.cloudflare.com/hc/en-us/';
export const TERMS_AND_CONDITIONS_PAGE = 'https://www.cloudflare.com/terms';
export const PRIVACY_POLICY_PAGE = 'https://www.cloudflare.com/security-policy';
export const CLOUDFLARE_ACCOUNT_PAGE = 'https://www.cloudflare.com/a/account/my-account';