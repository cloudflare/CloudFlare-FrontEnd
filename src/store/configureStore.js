import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import createLogger from 'redux-logger';
import rootReducer from '../reducers';
import { routerMiddleware } from 'react-router-redux';

export default function configureStore(history, initialState) {
  const router = routerMiddleware(history);

  const logger = createLogger({ collapsed: true });

  const createStoreWithMiddleware = applyMiddleware(
    thunkMiddleware,
    logger,
    router
  )(createStore);
  const store = createStoreWithMiddleware(rootReducer, initialState);

  return store;
}
