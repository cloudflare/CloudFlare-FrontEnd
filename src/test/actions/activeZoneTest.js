import {
  zoneSetActiveZone,
  asyncZoneSetActiveZone
} from '../../actions/activeZone';
import { ZONES_SET_ACTIVE_ZONE } from '../../constants/ActionTypes';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';

const mockStore = configureStore([thunk]);

describe('Active Zone Actions', () => {
  jest.mock('cf-util-http');

  describe('zoneSetActiveZone', () => {
    it('should match snapshot', () => {
      expect(zoneSetActiveZone({ id: 'id' })).toMatchSnapshot();
    });
  });

  describe('asyncZoneSetActiveZone', () => {
    it('should match snapshot for undefined zone.id', () => {
      let id = 'id';
      nock('https://api.cloudflare.com/client/v4')
        .filteringPath(function(path) {
          return '/';
        })
        .get('/')
        .reply(200, {});

      const store = mockStore();
      store.dispatch(asyncZoneSetActiveZone({}));
      expect(store.getActions()).toMatchSnapshot();
    });

    it('should match snapshot for valid zone.id', () => {
      const store = mockStore();
      store.dispatch(asyncZoneSetActiveZone({ id: 'id' }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
