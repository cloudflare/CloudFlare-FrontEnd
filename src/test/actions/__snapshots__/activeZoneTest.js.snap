// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Zone Actions asyncZoneSetActiveZone should match snapshot for undefined zone.id 1`] = `
Array [
  Object {
    "type": "ZONE_SET_ACTIVE_ZONE",
    "zone": Object {},
  },
]
`;

exports[`Active Zone Actions asyncZoneSetActiveZone should match snapshot for valid zone.id 1`] = `
Array [
  Object {
    "type": "ZONE_SET_ACTIVE_ZONE",
    "zone": Object {
      "id": "id",
    },
  },
  Object {
    "type": "DNS_RECORD_FETCH_LIST",
  },
  Object {
    "type": "ZONES_RAILGUNS_FETCH_ALL",
  },
  Object {
    "type": "PLUGIN_SETTINGS_FETCH",
  },
  Object {
    "type": "ZONE_FETCH_SETTINGS",
  },
  Object {
    "type": "ZONE_FETCH_ANALYTICS",
  },
]
`;

exports[`Active Zone Actions zoneSetActiveZone should match snapshot 1`] = `
Object {
  "type": "ZONE_SET_ACTIVE_ZONE",
  "zone": Object {
    "id": "id",
  },
}
`;
